mt_sat/mt_sat_wrapper.m

```diff
@@ -221,6 +237,24 @@ function mt_sat_wrapper(mtw_nii,pdw_nii,t1w_nii,mtw_jsn,pdw_jsn,t1w_jsn,varargin
     savejson('',provenance,[pwd filesep 'MTsat.json']);
 end
 
+% JSON file for dataset_description
+addDescription = struct();
+addDescription.Name = 'qMRLab Outputs';
+addDescription.BIDSVersion = '1.5.0';
+addDescription.DatasetType = 'derivative';
+addDescription.GeneratedBy.Name = 'qMRLab';
+addDescription.GeneratedBy.Version = qMRLabVer;
+addDescription.GeneratedBy.Container.Type = 'docker (if Docker{enabled = true})';
+addDescription.GeneratedBy.Container.Tag = 'qmrlab/minimal:2.4.1';
```

Comment:
Again, we could fetch this from the workflow.

Reply:
@agahkarakuzu I assigned `addDescription.GeneratedBy.Container.Type` the value given by `$workflow.containerEngine`. To set `addDescription.GeneratedBy.Container.Tag`, I tried `$workflow.container`, expecting the Docker image used to run workflow task, but it is always returning `[:]`

Reply:
Should not it be in `process.container`?  For multiple containers it is also a bit tricky. Let's see if you can access the name from the process scope.

Reply:
@agahkarakuzu it didn't work, `process.container` is also returning `[:]`

Reply:
Hmm, looks like `$workflow.container` returns a key-value pair per process

```
container | Docker image used to run workflow tasks. When more than one image is used it returns a map object containing [process name, image name] pair entries.

```

and you are not the only one run into this [issue](https://github.com/nextflow-io/nextflow/issues/1758). 

I saw `${workflow.container ?: '-'}` called in on completion reports by Nextflow, maybe give that a try?

Reply:
println "Container : ${workflow.container ?: '-'}"
println "Container : ${workflow.container}"
println "Container : $workflow.container"

let's see what they give.

Reply:
println "Container : ${workflow.container ?: '-'}" prints `Container: -`. It prints what is inside of ' '.
println "Container : ${workflow.container}" prints `Container: [:]` and the same,
println "Container : $workflow.container" prints `Container: [:]`

Reply:
Then there is nothing but complaining about this at https://github.com/nextflow-io/nextflow/issues/1758. It is present in the workflow report, but somehow we cannot access it. We can drop this for now, then implement it when nextflow resolves the issue.

Reply:
I agree. Do you want me to remove all traces referring to the `GeneratedBy.Container.Tag` field?

Reply:
@jvelazquez-reyes we can have a temporary solution by adding some vars at the config for now. If `null` pass, if not write in the json. 

You can add a comment in the config saying that this is a temporary solution, users are recommended to populate these fields if they are using containers.